qiskit/providers/aer/backends/aerbackend.py

```diff
@@ -313,6 +322,217 @@ def _run(self, qobj, job_id=''):
 
         return Result.from_dict(output)
 
+    def _is_dynamic(self, circuit):
+        if not isinstance(circuit, QuantumCircuit):
+            return False
+        for inst, _, _ in circuit.data:
+            if isinstance(inst, (WhileLoopOp, ForLoopOp, IfElseOp, BreakLoopOp, ContinueLoopOp)):
+                return True
+        return False
+
+    def _decompose_subcircuits(self, circuit, basis_gates=None):
```

Comment:
Can all these qasm3 compilation functions be moved back to a separate file, and just have a `compile_qasm3` function that is imported from there. It makes the class here very cluttered when all of these functions do not need to refer to the class instance.